megamek/src/megamek/common/Dropship.java

```diff
@@ -437,18 +437,19 @@ public double getCost(boolean ignoreAmmo) {
 
         // Transport Bays
         int baydoors = 0;
-        int bayCost = 0;
+        long bayCost = 0;
+        long quartersCost = 0;
         for (Bay next : getTransportBays()) {
             baydoors += next.getDoors();
-            if ((next instanceof MechBay) || (next instanceof ASFBay) || (next instanceof SmallCraftBay)) {
-                bayCost += 20000 * next.totalSpace;
-            }
-            if ((next instanceof LightVehicleBay) || (next instanceof HeavyVehicleBay)) {
-                bayCost += 10000 * next.totalSpace;
+            if (next.isQuarters()) {
+                quartersCost += next.getCost();
+            } else {
+                bayCost += next.getCost();
             }
         }
 
         costs[costIdx++] += bayCost + (baydoors * 1000);
```

Comment:
Apparently this would resolve a "potential" integer overflow (y'all got too many bay doors...).
```suggestion
        costs[costIdx++] += bayCost + (baydoors * 1000L);
```

Reply:
I was already on it.